core/sr-primitives/src/generic/unchecked_extrinsic.rs

```diff
@@ -103,9 +103,9 @@ where
 	fn decode<I: Input>(input: &mut I) -> Option<Self> {
 		// This is a little more complicated than usual since the binary format must be compatible
 		// with substrate's generic `Vec<u8>` type. Basically this just means accepting that there
-		// will be a prefix of u32, which has the total number of bytes following (we don't need
+		// will be a prefix of vector length (we don't need
 		// to use this).
-		let _length_do_not_remove_me_see_above: u32 = Decode::decode(input)?;
+		let _length_do_not_remove_me_see_above: Vec<()> = Decode::decode(input)?;
```

Comment:
Indeed, it shouldn't allocate, but still will iterate `n` times. 

> I'd prefer not to rely on internal details of how Vec is encoded.

Fair enough. The fact that it was different has lead to this issue in the first place. However, maybe the test that tests equivalence for some values would be enough?

Reply:
I've checked that `Vec<()>` does not do any capacity calculations. Capacity is always reported as `usize::max_value()` and `resize` function is just 6 ASM instructions in debug build, 1 inlined instruction in release build.

Reply:
Oh, interesting. That's actually makes sense, since it's ZST.

My observation was based on checking the wasm build, I wonder why it is generated this way.